.github/workflows/Runtime-upgrade-reset.yaml

```diff
@@ -1,17 +1,50 @@
 name: Upgrade-runtime-reset
 
-on: [workflow_dispatch]
+on: workflow_dispatch
 
 env:
   CARGO_TERM_COLOR: always
+  NIGHTLY_TOOLCHAIN_VERSION: ${{ secrets.NIGHTLY_TOOLCHAIN_VERSION }}
+
 jobs:
   upgrade-runtime:
     runs-on: ubuntu-latest
     steps:
-      - name: Download wasm runtime
-        run: wget https://github.com/gear-tech/gear/releases/download/build/gear_runtime.compact.compressed.wasm
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          components: clippy, rustfmt
+          target: wasm32-unknown-unknown
+
+      - name: Show specific nightly version
+        if: env.NIGHTLY_TOOLCHAIN_VERSION != ''
+        run: echo $NIGHTLY_TOOLCHAIN_VERSION | sed 's/-/ - /g'
+
+      - name: Install specific nightly toolchain
+        if: env.NIGHTLY_TOOLCHAIN_VERSION != ''
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-${{ env.NIGHTLY_TOOLCHAIN_VERSION }}
+          target: wasm32-unknown-unknown
```

Comment:
Pin requires also moving and renaming toolchain paths.

Reply:
Missed it, fixed.